src/qmapshack/gis/rte/router/brouter/CRouterBRouterSetup.cpp

```diff
@@ -726,17 +819,99 @@ void CRouterBRouterSetup::loadOnlineProfileFinished(QNetworkReply* reply)
     }
 }
 
+void CRouterBRouterSetup::setLocalBRouterJar(const QString &path)
+{
+    localBRouterJar = path;
+    const QString& jarFileName = QDir(localDir).absoluteFilePath(localBRouterJar);
+    const QStringList& classFiles = JlCompress::getFileList(jarFileName).filter(QRegularExpression(".*BRouter\\.class"));
+    if (!classFiles.isEmpty())
+    {
+        const QString& tmpFileName = JlCompress::extractFile(jarFileName,classFiles.first(),getDownloadDir().absoluteFilePath("BRouter.class"));
+        if (tmpFileName.endsWith("BRouter.class"))
+        {
+            QFile tmpFile(tmpFileName);
```

Comment:
Using `QTemporaryFile` with auto-remove set true would be much safer as you do not have to take care that the file is removed on every exit point.

Reply:
I guess QTemporaryFile will not work here as the physical file is created by JlCompress::extractFile before QFile is declared. QTemporaryFile with autoremoval is suitable when the creation of the physical file will be controlled by QTemporaryFile (by calling 'open'). I cannot see how I can make JlCompress return a QTemporaryFile without changing the library (which I do not intent to do)